File: Task 4/4a.py
'''4a.)Write a python program to create user defined exception.'''
class UA(Exception):
 pass
try:
    n=input('Enter Name:')
    a=int(input('Enter Age:'))
    if(a<0):
        raise TypeError("Enter age above 0")
    if(a<18):
        raise UA
    print(n,'is eligible to vote')
except TypeError as b:
        print(b)
except UA:
         print(n,"is not eligible to vote")
except ValueError:
         print("Enter a number in age above 0")

UA()

File: C Language/Task 4/4a.c
//a. Write a C Program check whether a given number is perfect number or not.
//Program:
#include<stdio.h>
void main()
 {
 int n,i,r=0;
 //printf("enter n:");
 scanf("%d",&n);
 for(i=1;i<=n;i++){
 if(n%i==0)
 r=r+i;
 }
 if(r==2*n)
 printf("%d is a perfect number",n);
 else
 printf("%d is not a perfect number",n);
 }
